exonum/src/helpers/config.rs

```diff
@@ -14,14 +14,48 @@
 
 //! Loading and saving TOML-encoded configurations.
 
-use std::path::Path;
-use std::io::{self, Read, Write};
+use std::path::{Path, PathBuf};
+use std::io::{Read, Write};
 use std::fs::{self, File};
 use std::error::Error;
+use std::fmt;
 
 use serde::{Serialize, Deserialize};
 use toml;
 
+#[derive(Debug)]
+struct DeserializeError {
+    path: PathBuf,
+    inner: toml::de::Error,
+}
+
+impl DeserializeError {
+    pub fn new(path: PathBuf, inner: toml::de::Error) -> Self {
+        Self { path, inner }
+    }
+}
+
+impl fmt::Display for DeserializeError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "Could not read toml config.\nPath: {}\nError: {}",
```

Comment:
Is the `\n` suitable for Windows?

Reply:
I don't have an opportunity to check, so I'll just remove them ;D

Reply:
Note that `writeln!` and `println!` also use `\n` on windows, and established tools like Cargo also just use `\n`: https://github.com/rust-lang/cargo/blob/d4e0b71c05df7768aabf95198cf606672788935d/src/cargo/lib.rs#L149

So, yes, `\n` works for windows just fine (unless you literally use `notepad.exe` in which case you have a more serious problem anyway), and we, imo, should stick to `\n` as well. 


(and a related piece of knowledge, forward slashes in paths also work on windows)

Reply:
Actually I don't think we need explicit newlines there. The output feels more dense and solid without them.

Reply:
Yeah, I don't think we need `\n` in this particular case! By the way, I think we can compress the message a bit more: `"Could not read {}: {}"`.